plutus-tx/src/PlutusTx/Foldable.hs

```diff
@@ -139,12 +139,14 @@ product = getProduct #. foldMap Product
 
 -- | Plutus Tx version of 'Data.Foldable.foldrM'.
 foldrM :: (Foldable t, Haskell.Monad m) => (a -> b -> m b) -> b -> t a -> m b
+-- See Note [Haskell.Monad]
 foldrM f z0 xs = foldl c Haskell.return xs z0
   where c k x z = f x z Haskell.>>= k
         {-# INLINE c #-}
```

Comment:
The existence of this function very much suggests that monads aren't only useful because of do-notation.

Reply:
I agree, this is quite suspect still. It's surprising that this works.